src/particleContainer/LinkedCellTraversals/NeutralTerritoryTraversal.h

```diff
@@ -0,0 +1,161 @@
+	 */
+	void rebuild(std::vector<CellTemplate>& cells, const std::array<unsigned long, 3>& dims, double cellLength[3],
+				 double cutoff, CellPairTraversalData* data) override {
+		CellPairTraversals<CellTemplate>::rebuild(cells, dims, cellLength, cutoff, data);
+		computeOffsets3D(cellLength, cutoff);
+		computeOffsets();
+	}
+
+	void traverseCellPairs(CellProcessor& cellProcessor) override;
+	void traverseCellPairsOuter(CellProcessor& cellProcessor) override;
+	void traverseCellPairsInner(CellProcessor& cellProcessor, unsigned stage, unsigned stageCount) override;
+
+	// NT traversal requires force exchange!
+	bool requiresForceExchange() const override { return true; }
+
+	// NT has no upper limit for number of cells in cutoff!
+	unsigned maxCellsInCutoff() const override { return std::numeric_limits<unsigned>::max(); }
+
+protected:
+	void processBaseCell(CellProcessor& cellProcessor, unsigned long baseIndex) const;
+
+	// All pairs that have to be processed when calculating the forces (excluding self)
+	std::vector<std::pair<long, long>> _cellPairOffsets;
+	std::vector<std::pair<std::array<long, 3>, std::array<long, 3>>> _offsets3D;
+
+private:
+	void computeOffsets();
+	void computeOffsets3D(const double cellLength[3], double cutoff);
+};
+
+template <class CellTemplate>
+void NeutralTerritoryTraversal<CellTemplate>::computeOffsets3D(const double cellLength[3], const double cutoff) {
+	_offsets3D.clear();
+	_cellPairOffsets.clear();
+	long num_x = std::ceil(cutoff / cellLength[0]);
+	long num_y = std::ceil(cutoff / cellLength[1]);
+	long num_z = std::ceil(cutoff / cellLength[2]);
+	for (long plate_x = 0; plate_x <= num_x; ++plate_x) {
+		// Start plate_y at 0, iff plate_x == 0.
+		long start_plate_y = plate_x == 0 ? 0 : -num_y;
+		for (long plate_y = start_plate_y; plate_y <= num_y; ++plate_y) {
+			// Start tower_z at 1, iff plate_x == 0 and plate_y == 0.
+			// This also prevents both tower and plate to be {0,0,0}.
+			long start_tower_z = (plate_x == 0 and plate_y == 0) ? 1 : -num_z;
+			for (long tower_z = start_tower_z; tower_z <= num_z; ++tower_z) {
+				std::array<long, 3> plate{plate_x, plate_y, 0};
+				std::array<long, 3> tower{0, 0, tower_z};
+				_offsets3D.emplace_back(tower, plate);
+			}
+		}
+	}
+}
+
+template <class CellTemplate>
+void NeutralTerritoryTraversal<CellTemplate>::computeOffsets() {
+	// Clear _cellPairOffsets!
+	_cellPairOffsets.clear();
+
+	using threeDimensionalMapping::threeToOneD;
+
+	// Dim array is int but we need it as long for some reason (copied from C08BasedTraversal)
+	std::array<long, 3> dims{};
+	for (int d = 0; d < 3; ++d) {
+		dims[d] = static_cast<long>(this->_dims[d]);
+	}
+
+	for (unsigned int i = 0; i < _offsets3D.size(); ++i) {
+		auto a = _offsets3D[i].first;
+		auto b = _offsets3D[i].second;
+
+		auto ax = std::get<0>(a);
+		auto ay = std::get<1>(a);
+		auto az = std::get<2>(a);
+
+		auto bx = std::get<0>(b);
+		auto by = std::get<1>(b);
+		auto bz = std::get<2>(b);
```

Comment:
```suggestion
	for (auto &[a, b] : _offsets3D) {
		auto [ax, ay, az] = a;
		auto [bx, by, bz] = b;
```
I would also argue for better names instead of a / b.

Reply:
that's even c++17 ;)